### ISSUE TITLE
Suggestions - Modernizing Editor's UI

### ISSUE THREAD

**Issue by jetrotal** [EasyRPG/Editor#182]
Overview
------------------
Inspired by the [RMUI's concept art ](https://finalbossblues.com/des/rpgmacker/rmui.html), I started drawing some some concept arts for EasyRPG's UI:

![image](https://user-images.githubusercontent.com/45118493/129250326-c57d80bc-1d1e-406b-9695-b5c7795ad3a7.png)

It's based on Google's Material Design.
There's a Qt framework for this kind of design, It's called [Qaterial](https://tinyurl.com/yfqz4h2g)

All Icons came from the https://materialdesignicons.com/
an open-source community that designs multiple icons based on the Material Design Style.

Header
-----------------------------
![image](https://user-images.githubusercontent.com/45118493/129242873-a9778fa2-a23e-489d-ba8a-10cc8c5c56ab.png)
The header would be simplified, I tried to move all the tools that used to be in the header, closer to relevant panels.

Tilesets Tab
----------------------------
![image](https://user-images.githubusercontent.com/45118493/129243785-ea1b89f4-d4a7-41d8-b348-06f33b762661.png)
The drawing tools would be placed vertically, imitating modern Drawing Apps. 
The zoom options are closer to the Zoom Tool.
And the map's tilesets can be changed on the fly.

Layers Tab
--------------------
![image](https://user-images.githubusercontent.com/45118493/129244241-2eb4e6ee-4230-4173-a2ac-eee86f8653f8.png)
I thought about naming the Layers "Terrain", "Decoration" and "Events", instead of "Lower", "Up" and "Events".

The "Decoration" layer comes with multiple indexes.
Those indexes would work as a replacement to blank Events that are just used for decoration, like in the tree decorations bellow:
![image](https://user-images.githubusercontent.com/45118493/129244649-bee15dcb-1bd9-4431-9dec-0f426543c4da.png)

Events Explorer Tab
--------------------
![image](https://user-images.githubusercontent.com/45118493/129254366-a81e02dc-1edb-4e92-be87-e80643281953.png)

A suggestion for a new tool, that let the User list all the events placed on each map, then making it easy to find, edit, copy or remove project's event. It would also include Commons Events.

![image](https://user-images.githubusercontent.com/45118493/129246185-2d31bcd2-b7b3-44fd-b398-60d553adab8f.png)
Those Transparent icons on the bottom, are a way to separate Blank Events from Events with commands, in case the Decoration Index suggestions can't be used.

Project Tree Tab
-------------------
![image](https://user-images.githubusercontent.com/45118493/129245497-ddee4fbd-2679-4ce9-83aa-45efb97f0c88.png)
Basically the current Map Tree Tab. With buttons for adding and removing maps

Map Properties Tab
--------------
![image](https://user-images.githubusercontent.com/45118493/129245785-aa80a68c-4fd4-4c8b-9e76-a2f503360da1.png)
A live version of the "Map Properties Window". 
"Save Map" and "Revert Map Changes" are also included.

------------------------------

Those are some thoughts I put over the Editor's UI. Maybe it's too far from what you guys have been planning. But I hope some ideas on it can be considered.

Thanks in advance!

**Comment by Ghabry** [comment]
This looks really cool. Maybe should really learn qtquick UI stuff compared to QWidgets. More design freedom :)

**Comment by jetrotal** [comment]
SplashScreeen
---------------------------

Inspired by RM2K Splash screen:
![image](https://user-images.githubusercontent.com/45118493/174405222-553ada83-10c7-4e1f-80d1-12fcd89f995b.png)

It consists on a file that displays random pics of community screenshots or fanarts

PNG Reference:
![image](https://user-images.githubusercontent.com/45118493/174405132-3103eb81-9bb7-40bf-8f81-b088a64e58c1.png)
![image](https://user-images.githubusercontent.com/45118493/174405070-da21e94a-6f19-4ea2-ac09-354deb718d03.png)

Editable: SVG File

![image](https://jetrotal.github.io/testing-stuff/SplashScreen.svg)
![image](https://jetrotal.github.io/testing-stuff/SplashScreenB.svg)

------------------

Icon
--------------------

I experimented reversing the color intensity and the overall shape, so the user could easily distinguish what is what
![image](https://user-images.githubusercontent.com/45118493/174405532-a35a1f2a-0650-47c7-b4bb-d2a717cb93d2.png) ![image](https://jetrotal.github.io/testing-stuff/icon.svg)
player / editor

**Comment by ryliejamesthomas** [comment]
The layers and events lists are really neat ideas
